src/providers/Asset.tsx

```diff
@@ -77,38 +78,47 @@ function AssetProvider({
     data: frePrice
   } = useQuery<FrePrice>(freQuery, {
     variables,
-    skip: true
+    skip: false
   })
   const {
     refetch: refetchPool,
     startPolling: startPollingPool,
     data: poolPrice
   } = useQuery<PoolPrice>(poolQuery, {
     variables,
-    skip: true
+    skip: false
   })
 
-  useEffect(() => {
-    if (!ddo || !variables) return
-    if (ddo.price.type === 'exchange') {
-      refetchFre(variables)
-      startPollingFre(refreshInterval)
-    } else {
-      refetchPool(variables)
-      startPollingPool(refreshInterval)
-    }
-  }, [ddo, variables])
+  // this is not working as expected, thus we need to fetch both pool and fre
+  // useEffect(() => {
+  //   if (!ddo || !variables || variables === '') return
+
+  //   if (ddo.price.type === 'exchange') {
+  //     refetchFre(variables)
+  //     startPollingFre(refreshInterval)
+  //   } else {
+  //     refetchPool(variables)
+  //     startPollingPool(refreshInterval)
+  //   }
+  // }, [ddo, variables])
 
   useEffect(() => {
-    if (!frePrice || frePrice.fixedRateExchanges.length === 0) return
+    if (
+      !frePrice ||
+      frePrice.fixedRateExchanges.length === 0 ||
+      price.type !== 'exchange'
+    )
+      return
     price.value = frePrice.fixedRateExchanges[0].rate
+    price.address = frePrice.fixedRateExchanges[0].id
```

Comment:
sidenote, I'm always not sure how React behaves when doing these object mutations with its state objects. Because of that, a better pattern is to modify anything in React state with the setState methods only. So in this case and all the similar cases this is the most React-safe way:

```js
setPrice((currentState) => ({
    ...currentState,
    value: frePrice.fixedRateExchanges[0].rate,
    address: frePrice.fixedRateExchanges[0].id
}))
```

We just don't run into issues here cause the `AssetProvider` is almost always mounted and unmounted before going to next asset

Reply:
cool finding.